File: 14-10-2024_Maximal_Score_After_Applying_K_Operations.java
import java.util.PriorityQueue;
import java.util.Collections;

class Solution {
    public long maxKelements(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());

        for (int num : nums) {
            pq.offer(num);
        }

        long score = 0;

        while (k > 0) {
            int ele = pq.poll();
            score += ele;
            pq.offer((int) Math.ceil(ele / 3.0));
            k--;
        }

        return score;
    }
}

File: 14-10-2024_Maximal_Score_After_Applying_K_Operations.py
class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
        # Convert nums to a max heap
        pq = [-num for num in nums]
        heapq.heapify(pq)

        score = 0

        for _ in range(k):
            ele = -heapq.heappop(pq)
            score += ele
            heapq.heappush(pq, -math.ceil(ele / 3))

        return score

